hubploy/auth.py

```diff
@@ -216,32 +249,66 @@ def cluster_auth_gcloud(deployment, project, cluster, zone, service_key):
     yield
 
 
-def cluster_auth_aws(deployment, project, cluster, zone, service_key):
+def cluster_auth_aws(deployment, project, cluster, zone, service_key=None, role=None):
     """
-    Setup AWS authentication with service_key
+    Setup AWS authentication with service_key or with a role
 
     This changes *global machine state* on what current kubernetes cluster is!
     """
-    # Get credentials from standard location
-    service_key_path = os.path.join(
-        'deployments', deployment, 'secrets', service_key
-    )
 
-    original_credential_file_loc = os.environ.get("AWS_SHARED_CREDENTIALS_FILE", None)
+    if not service_key and not role:
+        raise Exception('AWS authentication requires either a service key or the use of a role')
+
+    def update_kubeconfig():
```

Comment:
Can we put it outside the `if/else` below, instead of factoring it out into an inline function?

P.S. I <3 inline functions :D

Reply:
Not sure what you mean here.  It's already outside the `if/else` block.  Or maybe this comment was written before I moved that?  Please double check once I push my most recent changes.

Reply:
Currently, this is a function that's called in the last lines of the two branches of the if/else, but nowhere else. This means, when the if/else exits, *no matter what*, the contents of this function would be called. We can simplify this by removing the function, and just having this subprocess call happen after the `if/else`, in like 306, just before the 'yield'

Reply:
Good point.